server/service/src/processors/transfer/requisition/mod.rs

```diff
@@ -75,13 +75,22 @@ pub(crate) fn process_requisition_transfers(
     let active_stores =
         ActiveStoresOnSite::get(&ctx.connection).map_err(Error::GetActiveStoresOnSiteError)?;
 
+    let name_ids = active_stores.name_ids();
+    let name_link_repo = NameLinkRowRepository::new(&ctx.connection);
+    let name_link_ids: Vec<String> = name_link_repo
+        .find_many_by_name_ids(name_ids.as_slice())
+        .unwrap() // MERGE: handle error
+        .iter()
+        .map(|r| r.id.clone())
+        .collect();
+
     let changelog_repo = ChangelogRepository::new(&ctx.connection);
     let key_value_store_repo = KeyValueStoreRepository::new(&ctx.connection);
     // For transfers, changelog MUST be filtered by records where name_id is active store on this site
     // this is the contract obligation for try_process_record in ProcessorTrait
     let filter = ChangelogFilter::new()
         .table_name(ChangelogTableName::Requisition.equal_to())
-        .name_id(EqualFilter::equal_any(active_stores.name_ids()))
+        .name_id(EqualFilter::equal_any(name_link_ids.clone()))
```

Comment:
Since you already do the join the ChangelogRepository I would change it really query by `name_id` and not by `name_link_id` (I believe thats what we do anywhere else). Then this code becomes much easier as well. For example:

```Rust
.name_id(EqualFilter::equal_any(active_stores.name_ids())
```

And you don't need the name_link_row method...

Reply:
TRUE 31f88eac3c0ea466f787bee28f8de101b2cad8bb thanks